### PR TITLE
Datetime

### PR DESCRIPTION
**Pull request by zupercool**

### CODE REVIEW

File: tests/stdlib/test_datetime.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
+from unittest import expectedFailure
+
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 1: (F401) 'unittest.expectedFailure' imported but unused

File: tests/stdlib/test_datetime.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+from ..utils import TranspileTestCase
+
+# class DateTimeModuleTests(TranspileTestCase):
+
+# class TimeDeltaTests(TranspileTestCase):
+
+class DateTimeTests(TranspileTestCase):
+
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 1: (E302) expected 2 blank lines, found 1

File: tests/stdlib/test_datetime.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    def test_creation(self):
+       self.assertCodeExecution("""
+           from datetime import datetime
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 8: (E111) indentation is not a multiple of four

File: tests/stdlib/test_datetime.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+           print(datetime(1,2,3))
+           print(datetime(1,2,day=3))
+           print(datetime(1,month=2,day=3))
+           print(datetime(year=1,month=2,day=3))
+
+           print(datetime(11,12,13))
+           print(datetime(111,12,13))
+           print(datetime(1111,12,13))
+           
+           print(datetime(1,1,1,0,0,0))
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_datetime.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+           print(datetime(1,1,1,0,0,0,0))
+           print(datetime(9999,12,31,23,59,59))
+           print(datetime(9999,12,31,23,59,59,999999))
+           
+           print(datetime(1,2,3,4,5,6,7))
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_datetime.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+           """)
+
+    # The following test will not succeed because the microseconds will differ.
+    # def test_today(self):
+    #     self.assertCodeExecution("""
+    #         from datetime import datetime
+    #         print(datetime.today())
+    #     """)
+    
+    def test_date(self):
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_datetime.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+            print (datetime(1993,5,17).year)
+            print (datetime(1993,5,17).month)
+            print (datetime(1993,5,17,20,30,12,34).hour)
+            print (datetime(1993,5,17,20,30,12,34).minute)
+            print (datetime(1993,5,17,20,30,12,34).second)
+            print (datetime(1993,5,17,20,30,12,34).microsecond)
+        """)
+
+    
+    def test_year_too_large(self):
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_datetime.py [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        self.assertCodeExecution("""
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 5: (E303) too many blank lines (2)

File: tests/stdlib/test_datetime.py [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+        from datetime import datetime
+        try:
+            datetime(19999, 10, 11)
+        except ValueError as err:
+            print(err)
+        
+        """)
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_datetime.py [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+    def test_year_too_large(self):
+        self.assertCodeExecution("""
+        from datetime import datetime
+        try:
+            datetime(19999, 10, 11)
+        except ValueError as err:
+            print(err)
+        
+        """)    
+
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 13: (W291) trailing whitespace

File: tests/stdlib/test_datetime.py [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+    def test_year_too_small(self):
+        self.assertCodeExecution("""
+        from datetime import datetime
+        try:
+            datetime(0, 10, 11)
+        except ValueError as err:
+            print(err)
+        
+        """)
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_datetime.py [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+    def test_year_too_small(self):
+        self.assertCodeExecution("""
+        from datetime import datetime
+        try:
+            datetime(0, 10, 11)
+        except ValueError as err:
+            print(err)
+        
+        """)  
+
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 13: (W291) trailing whitespace

File: tests/stdlib/test_datetime.py [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+    def test_month_too_large(self):
+        self.assertCodeExecution("""
+        from datetime import datetime
+        try:
+            datetime(14, 41, 11)
+        except ValueError as err:
+            print(err)
+        
+        """)
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_datetime.py [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+    def test_month_too_large(self):
+        self.assertCodeExecution("""
+        from datetime import datetime
+        try:
+            datetime(14, 41, 11)
+        except ValueError as err:
+            print(err)
+        
+        """)  
+
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 13: (W291) trailing whitespace

File: tests/stdlib/test_datetime.py [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+        self.assertCodeExecution("""
+        from datetime import datetime
+        try:
+            datetime(14, 41, 11)
+        except ValueError as err:
+            print(err)
+        
+        """)  
+        
+    def test_month_too_small(self):
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_datetime.py [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+            self.assertCodeExecution("""
+        from datetime import datetime
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 13: (E117) over-indented

File: tests/stdlib/test_datetime.py [hunk 17]
```diff
@@ -0,0 +1,124 @@
...
+        try:
+            datetime(0, 0, 11)
+        except ValueError as err:
+            print(err)
+        
+        """)
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_datetime.py [hunk 18]
```diff
@@ -0,0 +1,124 @@
...
+    def test_month_too_small(self):
+            self.assertCodeExecution("""
+        from datetime import datetime
+        try:
+            datetime(0, 0, 11)
+        except ValueError as err:
+            print(err)
+        
+        """)       
+
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 13: (W291) trailing whitespace

File: tests/stdlib/test_datetime.py [hunk 19]
```diff
@@ -0,0 +1,124 @@
...
+            self.assertCodeExecution("""
+        from datetime import datetime
+        try:
+            datetime(0, 0, 11)
+        except ValueError as err:
+            print(err)
+        
+        """)       
+    
+    # def test_creation_invaild_type_str(self):
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_datetime.py [hunk 20]
```diff
@@ -0,0 +1,124 @@
...
+    #         self.assertCodeExecution("""
+    #     from datetime import datetime
+    #     try:
+    #         datetime("ss", 0, 11)
+    #     except ValueError as err:
+    #         print(err)
+
+    #     """) 
+
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 15: (W291) trailing whitespace

File: tests/stdlib/test_datetime.py [hunk 21]
```diff
@@ -0,0 +1,124 @@
...
+    # def test_creation_invaild_type_float(self):
+    #         self.assertCodeExecution("""
+    #     from datetime import datetime
+    #     try:
+    #         datetime(14.0, 0, 11)
+    #     except ValueError as err:
+    #         print(err)
+
+    #     """)
\ No newline at end of file
```
Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 15: (W292) no newline at end of file

Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 15: (W292) no newline at end of file

Comment (brutusthebee) [inline tests/stdlib/test_datetime.py]: At column 15: (W292) no newline at end of file

File: tests/stdlib/test_math.py [hunk 22]
```diff
@@ -0,0 +1,63 @@
+from unittest import expectedFailure
+
+from ..utils import TranspileTestCase
+
+class MathModuleTests(TranspileTestCase):
+    def test_floor(self):
```
Comment (brutusthebee) [inline tests/stdlib/test_math.py]: At column 1: (E302) expected 2 blank lines, found 1

File: tests/stdlib/test_math.py [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+        self.assertCodeExecution("""
+            import math
+            print(math.floor(7.35))
+            print(math.floor(True))
+            print(math.floor(False))
+            print(math.floor(-1))
+            print(math.floor(float(32.3)))
+            """)
+    def test_ceil(self):
+        self.assertCodeExecution("""
```
Comment (brutusthebee) [inline tests/stdlib/test_math.py]: At column 5: (E301) expected 1 blank line, found 0

File: tests/stdlib/test_math.py [hunk 24]
```diff
@@ -0,0 +1,63 @@
...
+
+    def test_fabs(self):
+        self.assertCodeExecution("""
+            import math
+            print(math.fabs(-1))
+            print(math.fabs(-2.0))
+            print(math.fabs(0))
+            """)
+        
+    @expectedFailure
```
Comment (brutusthebee) [inline tests/stdlib/test_math.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_math.py [hunk 25]
```diff
@@ -0,0 +1,63 @@
...
+    def test_ceil_expectedFailure(self):
+        self.assertCodeExecution("""
+            import math
+            print(math.ceil("hej"))
+            print(math.ceil([]))
+            """)
+    @expectedFailure
+    def test_fabs_expectedFailure(self):
```
Comment (brutusthebee) [inline tests/stdlib/test_math.py]: At column 5: (E301) expected 1 blank line, found 0

File: tests/stdlib/test_math.py [hunk 26]
```diff
@@ -0,0 +1,63 @@
...
+
+    @expectedFailure
+    def test_floor_expectedFailure(self):
+        self.assertCodeExecution("""
+            import math
+            print(math.floor("string input"))
+            print(math.floor([]))
+            """)
+            
+    @expectedFailure
```
Comment (brutusthebee) [inline tests/stdlib/test_math.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_math.py [hunk 27]
```diff
@@ -0,0 +1,63 @@
...
+    def test_floor_expectedFailure2(self):
+        self.assertCodeExecution("""
+            import math
+            print(math.floor())
+            """)
+           
+
```
Comment (brutusthebee) [inline tests/stdlib/test_math.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_math.py [hunk 28]
```diff
@@ -0,0 +1,63 @@
...
+            
+    @expectedFailure
+    def test_floor_expectedFailure2(self):
+        self.assertCodeExecution("""
+            import math
+            print(math.floor())
+            """)
+           
+            
+
```
Comment (brutusthebee) [inline tests/stdlib/test_math.py]: At column 1: (W293) blank line contains whitespace

File: tests/stdlib/test_math.py [hunk 29]
```diff
@@ -0,0 +1,63 @@
...
+    @expectedFailure
+    def test_floor_expectedFailure2(self):
+        self.assertCodeExecution("""
+            import math
+            print(math.floor())
+            """)
+           
+            
+ 
\ No newline at end of file
```
Comment (brutusthebee) [inline tests/stdlib/test_math.py]: At column 1: (W391) blank line at end of file

Comment (brutusthebee) [inline tests/stdlib/test_math.py]: At column 1: (W293) blank line contains whitespace

File: tests/datatypes/test_datetime.py [hunk 30]
```diff
@@ -0,0 +1,21 @@
+from unittest import expectedFailure
+
+from ..utils import TranspileTestCase
+
+class DateTimeModuleTests(TranspileTestCase):
+    def test_ceil(self):
```
Comment (brutusthebee) [inline tests/datatypes/test_datetime.py]: At column 1: (E302) expected 2 blank lines, found 1

File: tests/datatypes/test_datetime.py [hunk 31]
```diff
@@ -0,0 +1,21 @@
...
+            import datetime
+            print(datetime.timeDelta(4.5))
+            """)
+
+    def test_timeDelta(self):
+        self.assertCodeExecution("""
+            import datetime
+            """)
+            
+    @expectedFailure
```
Comment (brutusthebee) [inline tests/datatypes/test_datetime.py]: At column 1: (W293) blank line contains whitespace

File: tests/datatypes/test_datetime.py [hunk 32]
```diff
@@ -0,0 +1,21 @@
...
+    def test_datetime(self):
+        self.assertCodeExecution("""
+            import datetime
+            """)
\ No newline at end of file
```
Comment (brutusthebee) [inline tests/datatypes/test_datetime.py]: At column 17: (W292) no newline at end of file

**Comment by phildini** [comment]
Hi there! It looks like this PR might be dead, so we're closing it for now. Feel free to re-open it if you'd like to continue, or think about directing your efforts to https://github.com/beeware/briefcase or https://github.com/beeware/toga. Both of these have more active development right now. 😄

**Pull request closed without merging**